source/MaterialXView/ShaderValidators/Osl/OslValidator.cpp

```diff
@@ -27,6 +29,76 @@ OslValidator::~OslValidator()
 
 void OslValidator::initialize()
 {
+    ShaderValidationErrorList errors;
+    const std::string errorType("OSL initialization error.");
+    if (_oslIncludePathString.empty())
+    {
+        errors.push_back("OSL validation include path is empty.");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+    if (_oslTestShadeExecutable.empty() && _oslCompilerExecutable.empty())
+    {
+        errors.push_back("OSL validation executables not set.");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+}
+
+void OslValidator::shadeOSL(const std::string& shaderName, const std::string& outputName)
+{
+    // If no command and include path specified then skip checking.
+    if (_oslTestShadeExecutable.empty() || _oslIncludePathString.empty())
+    {
+        return;
+    }
+
+    // Set output image name. 
+    std::string outputFileName = shaderName + ".png";
+
+    // Use a known error file name to check
+    std::string errorFile(shaderName + "_render_errors.txt");
+    const std::string redirectString(" 2>&1");
+
+    std::string command(_oslTestShadeExecutable);
+    command += " " + shaderName;
+    command += " -o " + outputName + " " + outputFileName;
+    command += " -g 256 256";
+    command += " > " + errorFile + redirectString;
+
+    int returnValue = std::system(command.c_str());
+
+    // There is no "silent" or "quiet" mode for testshade so we must parse the lines
```

Comment:
Need to manually iterate to ignore any added newlines CR, LFs and a successful output line as there is no "quit" mode like for oslc which does not output anything if it's successful.